icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -2786,18 +2838,24 @@ RBBILineMonkey::RBBILineMonkey() :
     fSets->addElement(fOP30, status); classNames.push_back("fOP30");
     fSets->addElement(fCP30, status); classNames.push_back("fCP30");
     fSets->addElement(fExtPictUnassigned, status); classNames.push_back("fExtPictUnassigned");
+    fSets->addElement(fAK, status); classNames.push_back("fAK");
+    fSets->addElement(fAP, status); classNames.push_back("fAP");
+    fSets->addElement(fAS, status); classNames.push_back("fAS");
+    fSets->addElement(fVF, status); classNames.push_back("fVF");
+    fSets->addElement(fVI, status); classNames.push_back("fVI");
 
-    const char *rules =
-            "((\\p{Line_Break=PR}|\\p{Line_Break=PO})(\\p{Line_Break=CM}|\\u200d)*)?"
-            "((\\p{Line_Break=OP}|\\p{Line_Break=HY})(\\p{Line_Break=CM}|\\u200d)*)?"
-            "((\\p{Line_Break=IS})(\\p{Line_Break=CM}|\\u200d)*)?"
-            "\\p{Line_Break=NU}(\\p{Line_Break=CM}|\\u200d)*"
-            "((\\p{Line_Break=NU}|\\p{Line_Break=IS}|\\p{Line_Break=SY})(\\p{Line_Break=CM}|\\u200d)*)*"
-            "((\\p{Line_Break=CL}|\\p{Line_Break=CP})(\\p{Line_Break=CM}|\\u200d)*)?"
-            "((\\p{Line_Break=PR}|\\p{Line_Break=PO})(\\p{Line_Break=CM}|\\u200d)*)?";
 
-    fNumberMatcher = new RegexMatcher(
-        UnicodeString(rules, -1, US_INV), 0, status);
+    UnicodeString CMx {uR"([[\p{Line_Break=CM}]\u200d])"};
+    UnicodeString rules;
+    rules = rules + u"((\\p{Line_Break=PR}|\\p{Line_Break=PO})(" + CMx + u")*)?"
```

Comment:
optional / simpler:
```
UnicodeString rules =
    u"..."
    u"..."
    u"...";
```
using C++ string literal concatenation in the compiler.

Reply:
That does not work: CMx is not a literal (it would work if we made it a macro; do we want that?).